File: go/compressible_test.go
package compressible

import "testing"

func TestExt_Compressible(t *testing.T) {
	cases := []string{
		"js", "css", "html", "json", "svg", "wasm",
		"b3dm", "i3dm", "pnts", "cmpt", "subtree", "terrain", "bin",
	}
	for _, c := range cases {
		if !Ext(c) {
			t.Errorf("Ext(%q) = false, want true", c)
		}
	}
}

func TestExt_NotCompressible(t *testing.T) {
	cases := []string{
		"gz", "zip", "7z", "jpg", "jpeg", "png", "mp4", "mp3",
		"svgz", "psd", "vmdk", "tar", "ova",
	}
	for _, c := range cases {
		if Ext(c) {
			t.Errorf("Ext(%q) = true, want false", c)
		}
	}
}

func TestExt_Normalization(t *testing.T) {
	cases := []string{".JS", "JS", ".js", "js", "Js", ".Json"}
	for _, c := range cases {
		if !Ext(c) {
			t.Errorf("Ext(%q) = false, want true", c)
		}
	}
}

func TestExt_Empty(t *testing.T) {
	if Ext("") {
		t.Error("Ext(\"\") = true, want false")
	}
	if Ext(".") {
		t.Error("Ext(\".\") = true, want false")
	}
}

func TestPath(t *testing.T) {
	tt := []struct {
		in   string
		want bool
	}{
		{"foo.js", true},
		{"a/b/c.html", true},
		{"/abs/path/file.JSON", true},
		{"image.png", false},
		// Only the last extension matters.
		{"archive.tar.gz", false},
		{"data.json.gz", false},
		{"noext", false},
		{"", false},
	}
	for _, tc := range tt {
		if got := Path(tc.in); got != tc.want {
			t.Errorf("Path(%q) = %v, want %v", tc.in, got, tc.want)
		}
	}
}

func TestExtensions_SortedAndCopy(t *testing.T) {
	a := Extensions()
	if len(a) == 0 {
		t.Fatal("Extensions() empty")
	}
	for i := 1; i < len(a); i++ {
		if a[i-1] >= a[i] {
			t.Fatalf("Extensions() not sorted at %d: %q >= %q", i, a[i-1], a[i])
		}
	}
	// Mutating the result must not affect subsequent calls.
	a[0] = "ZZZZ"
	b := Extensions()
	if b[0] == "ZZZZ" {
		t.Error("Extensions() returned shared slice")
	}
}

File: js/test/compressible.test.ts
import { describe, expect, it } from "vitest";
import { extensions, isCompressible, isCompressiblePath } from "../src/index.js";

describe("isCompressible", () => {
  it("returns true for known compressible extensions", () => {
    for (const e of ["js", "css", "html", "json", "svg", "wasm", "b3dm", "i3dm", "pnts", "cmpt", "subtree", "terrain", "bin"]) {
      expect(isCompressible(e), e).toBe(true);
    }
  });

  it("returns false for already-compressed / binary formats", () => {
    for (const e of ["gz", "zip", "7z", "jpg", "jpeg", "png", "mp4", "mp3", "svgz", "psd", "vmdk", "tar", "ova"]) {
      expect(isCompressible(e), e).toBe(false);
    }
  });

  it("normalizes case and leading dot", () => {
    for (const e of [".JS", "JS", ".js", "Js", ".Json"]) {
      expect(isCompressible(e), e).toBe(true);
    }
  });

  it("returns false for empty input", () => {
    expect(isCompressible("")).toBe(false);
    expect(isCompressible(".")).toBe(false);
  });
});

describe("isCompressiblePath", () => {
  it("inspects only the last extension", () => {
    expect(isCompressiblePath("foo.js")).toBe(true);
    expect(isCompressiblePath("a/b/c.html")).toBe(true);
    expect(isCompressiblePath("/abs/path/file.JSON")).toBe(true);
    expect(isCompressiblePath("image.png")).toBe(false);
    expect(isCompressiblePath("archive.tar.gz")).toBe(false);
    expect(isCompressiblePath("data.json.gz")).toBe(false);
    expect(isCompressiblePath("noext")).toBe(false);
    expect(isCompressiblePath("")).toBe(false);
    expect(isCompressiblePath(".bashrc")).toBe(false);
  });

  it("handles Windows-style separators", () => {
    expect(isCompressiblePath("C:\\src\\foo.js")).toBe(true);
  });
});

describe("extensions", () => {
  it("is a non-empty set", () => {
    expect(extensions.size).toBeGreaterThan(0);
  });
});
